src/components/TreeGrid.ts

```diff
@@ -52,7 +52,9 @@ type TreeGridProps<T extends RowContents> = {
   /** the initial rows to display */
   readonly initialRows: T[];
   /** Given a row, return its children. This is only called if row.hasChildren is true. */
-  readonly getChildren: (row: T) => Promise<T[]>;
+  readonly getChildren: (datasetId: string, row: T) => Promise<T[]>;
```

Comment:
I don't think this is correct, I'll have to take a closer look but there should be a way to use `datasetId` in the dataset builder code without having to add it to the `TreeGrid` API.